### PR TITLE
Enable no-floating-promises eslint check as warning

### PR DESCRIPTION
**Pull request by dhuebner**
Enabled as warning because of #453

Overall 6 Problems detected by this rule.

### CODE REVIEW

**Comment by dhuebner** [comment]
Feel free to merge after accepted! 

Should we add a label: **accept2ship**  ? :)

File: packages/langium/tsconfig-eslint.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (spoenemann) [inline packages/langium/tsconfig-eslint.json:1]: Why did you add this file? What is is used for?

Comment (dhuebner) [reply packages/langium/tsconfig-eslint.json:1]: It is used in .eslintrc.json in `project` section to be when checking the `test` folder. Otherwise running `eslint src test` produces errors like `Can't read tsconfig.json` wenn analyzing the test files.

Comment (dhuebner) [reply packages/langium/tsconfig-eslint.json:1]: One could probably reduce the content to this one:
```json
{
  "extends": "./tsconfig.json",
  "include": [
    "test/**/*.ts"
  ]
}

```

Comment (spoenemann) [reply packages/langium/tsconfig-eslint.json:1]: Oh I see, it's because of this:

> Error: Error while loading rule '@typescript-eslint/no-floating-promises': You have used a rule which requires parserServices to be generated. You must therefore provide a value for the "parserOptions.project" property for @typescript-eslint/parser.

Comment (dhuebner) [reply packages/langium/tsconfig-eslint.json:1]: Yes, this rule needs Types information, otherwise it is almost useless

File: .eslintrc.json [hunk 2]
```diff
@@ -7,7 +7,15 @@
     ],
     "parserOptions": {
         "ecmaVersion": 6,
-        "sourceType": "module"
+        "sourceType": "module",
+        "project": [
+            "./tsconfig.json",
+            "./tsconfig-eslint.json",
```
Comment (spoenemann) [inline .eslintrc.json:13]: Now I see this error in the editor:

> Parsing error: Cannot read file '/workspace/langium/tsconfig-eslint.json'.

File: .eslintrc.json [hunk 3]
```diff
@@ -7,7 +7,15 @@
...
+            "./packages/generator-langium/tsconfig.json",
+            "./packages/langium-cli/tsconfig.json",
+            "./packages/langium-sprotty/tsconfig.json",
+            "./packages/langium-vscode/tsconfig.json"
```
Comment (spoenemann) [inline .eslintrc.json:17]: Are these necessary? It didn't seem to make a difference when I tried removing these lines.

Comment (dhuebner) [reply .eslintrc.json:17]: I just noticed, that after restarting ESlint server linting in editor fails :( 
We will probably need to try something else. It seem to behave differently wenn running eslint in CLI and using it in the editor

File: .eslintrc.json (hunk 2 above)
Comment (dhuebner) [reply .eslintrc.json:13]: Yes, this is kind of a nightmare. I found any way to get everything (IDE, CLI and Tests) working at once. 
Is it possible to move the `test` folder inside the `src` and then exclude it from packaging?

**Comment by spoenemann** [comment]
@dhuebner shall we close this?

**Comment by dhuebner** [comment]
@spoenemann 
Yes, let close it. But the check is still disabled, I didn't managed to get both vscode tooling and CLI working at the same time. :(
